### ISSUE TITLE
[Feature Request] Accomodate User Provided Fallback

### ISSUE THREAD

**Issue by dakanji** [acidanthera/bugtracker#858]
Would you consider checking for, and executing, a `fallback.efi` file in /EFI/BOOT if one is present and only proceeding to load OpenCore if not?

**Comment by vit9696** [comment]
I am not positive it is a reasonable addition, so please explain the motivation for us to understand the reasons behind this suggestion first.

**Comment by dakanji** [comment]
A long shot request really and I anticipate there is not likely to be much appetite for this.
Just wanted to be able to divert OC boot process into something else if I want to.

Not super important ... Just a nice to have option.

**Comment by vit9696** [comment]
I do not think we can do this unless you explain the reasons or at least provide some use case.

**Comment by dakanji** [comment]
Basically, I am preferring Refind as my hub for multi booting.

I know OC has OpenControl under development but on a Mac, I am testing and really liking having Refind be what boots up and from where I can branch out to various OS types including Legacy Windows. OpenCore is used purely for MacOS.

Also, I am able to have two instances of OpenCore available in Refind. One with Video Hardware Acceleration and One WIthout as some older macOS versions can have issues when running with HWAcceleration.

So My choices in Refind are:
* OpenCore with HWAcceleration
* OpenCore without HWAcceleration
* Windows
* Linux

With such a fallback that will actually boot Refind instead, I can keep Refind and OpenCore in the same EFI partition and given the quirky Apple implementation of the EFI spec that invariably leads to OpenCore's BOOTx64 efi file getting booted, I can break out of this.

As said though, not super critical as there are ways around this but it would just give a seamless experience and if the fallback is not there, then everything just works as normal.

**Comment by vit9696** [comment]
> I know OC has OpenControl under development but on a Mac, I am testing and really liking having Refind be what boots up and from where I can branch out to various OS types including Legacy Windows. OpenCore is used purely for MacOS.

We do not support legacy Windows and see no reasons to use it on any hardware including MacPro. If absolutely necessary you can still boot legacy Windows from OpenCore if you add a tool that launches MBR loaded OS, however you will need programming knowledge to write it. I can imagine such a tool appearing in OC in some distant future, but it would have been easier if somebody contributed it actually.

> Also, I am able to have two instances of OpenCore available in Refind. One with Video Hardware Acceleration and One WIthout as some older macOS versions can have issues when running with HWAcceleration.

Your solution to this problem is not valid. I have already explained to @startergo and others that you are doing it wrong. Basically there is no single reason to introduce a second configuration to disable HEVC encoding. OpenCore supports a thing named `MaxKernel`. Using this you can add kernel patch of the relevant AMD hardware kext (e.g. AMDRadeonX4000.kext) or a plist-only kext with removed IOGVAHEVCDecode. Watch out that unless `disable-gva-support` is specified, WEG will inject HEVC decoding properties into X4000 GPUs. Let me know if you need further clarification or assistance with WEG configuration. I believe it is pretty much obvious from the description, however.

> With such a fallback that will actually boot Refind instead, I can keep Refind and OpenCore in the same EFI partition and given the quirky Apple implementation of the EFI spec that invariably leads to OpenCore's BOOTx64 efi file getting booted, I can break out of this.

OpenCore's BOOTx64 is optional, and can be renamed/removed/moved wherever.

All in all, I believe it makes no sense. Closing as invalid.

**Comment by dakanji** [comment]
Fair enough ... I knew it was a long shot.
I am still able to get what I want without this.

Thanks for considering.

**Comment by dakanji** [comment]
> OpenCore's BOOTx64 is optional, and can be renamed/removed/moved wherever.
BTW, renaming OpenCore's BOOTx64 can lead to a recovery boot loop if `Reset NVRAM` is used.

I tried this once and I kept getting booted back into recovery on a Mac.

I had to get into the EFI via Terminal in Recovery to rename it back to the standard name to break the loop.

I believe that once `Reset NVRAM` is used, it tries to load BOOTx64 and when it doest find it, kicks things into recovery. Selecting a start disk there is no use as once you restart, it still tries to get to that file.

**Comment by vit9696** [comment]
Well, if you reset NVRAM, your firmware will indeed not be able to see the OpenCore image. We are actually thinking of something to workaround this in the future, but I am not sure when it is going to land.

**Comment by dakanji** [comment]
> We are actually thinking of something to workaround this in the future

Noted. 

It was a potentially very sticky situation as many on Legacy MacPros use new GPUs that may not be able to show the recovery screen on such machines to be able to get into Terminal for a fix.

Advice is to always keep a copy of an older GPU with Apple EFI for such situations but I am sure some don't have one of those at hand ... especially if they bought the cMP with the new GPU already in place as opposed to upgrading things themselves.

Still a bit unclear as to what happens when the `Reset NVRAM` feature is used as I thought that since it triggers a reboot, it would start the boot process from scratch as opposed to basically trying to pick up where it left off.

**Comment by vit9696** [comment]
`Reset NVRAM` erases NVRAM including firmware boot options. That's all of it. So it will obviously erase your blessed entry.

**Comment by dakanji** [comment]
I thought so but what seemed to happen was:
* I had renamed BOOTx64 to OC_BOOTx64
* After one reboot where a disk on a PCIE Slot did not load, I used Reset NVRAM as a proxy for Reboot
* Machine rebooted into recovery which is consistent with firmware boot options being erased and no valid boot efi found
* I selected start disk  in recovery and clicked the restart button
* Typically it should boot to the start disk but I got bounced into recovery again.
* I think I went through another such cycle before deciding to mount the EFI and edit the file name back to BOOTx64
* On restarting, I got booted into the selected Start Disk.

**Comment by dakanji** [comment]
Probably a crazy thought but could it be that the `Reset NVRAM` command persisted and was being reapplied on each restart ... basically removing the selected start disk each time?

**Comment by dakanji** [comment]
> We do not support legacy Windows and see no reasons to use it on any hardware including MacPro. If absolutely necessary you can still boot legacy Windows from OpenCore if you add a tool that launches MBR loaded OS, however you will need programming knowledge to write it. I can imagine such a tool appearing in OC in some distant future, but it would have been easier if somebody contributed it actually.

For anyone with a Classic MacPro 3,1 (Early 2008) reading this, note that you cannot install UEFI Windows and that the only option is Legacy Windows.

Pending the development of the tool that launches MBR loaded OS mentioned, your only option if wanting to run Windows is to chain load from a rEFIt Variant as as rEFInd.

Those on later Classic MacPro can use OpenCore to run UEFI Windows.

**Comment by vit9696** [comment]
Could you please explain why installing Windows in UEFI mode is not possible? Just to make it more clear, Windows installs in UEFI mode just fine on my old computer with Pentium 4 dating back to 2005. Why is MacPro any different?

**Comment by dakanji** [comment]
> Could you please explain why installing Windows in UEFI mode is not possible?

[I read it on the Internet](https://forums.macrumors.com/threads/mac-pro-3-1-nvme-support-upgrade-guide-questions.2194878/post-28397420) :wink:

**Comment by vit9696** [comment]
Well, that does not feel like a proof to me to be honest. Is there any confirmation?

**Comment by vit9696** [comment]
In fact the exact topic you linked shows that one can install Windows in UEFI mode on MacPro2,1 just fine, and that pretty much feels like it really should be.

**Comment by dakanji** [comment]
You are right ... I suppose I should rephrase the "you cannot" part 
Still seems like chain loading is easier and works well.

I know it could make your stomach turn but I have mine set up to run 4 different OC instances.
Anyway, don't want to distract you on this matter.

Thanks for the awesomeness of OpenCore

**Comment by vit9696** [comment]
Well, I indeed cannot say that trying to use OpenCore in a way it was not meant to be will get you praise here.

We have already discovered that HEVC works fine in 10.13 (at least in the latest update), and Windows seems to install just fine in UEFI mode as well. We also believe that our UI is visually far better and more native than rEFInd. I understand that's more about habits and that it may not be easy to change them, but we cannot invest in the habits we do not share ourselves.
